torch/_prims_common/__init__.py

```diff
@@ -491,7 +494,7 @@ def __eq__(self, other):
 
     expected_stride = 1
     for length, stride in lengths_and_strides:
-        if guard_size_oblivious(length == 1):
+        if guard_or_false(length == 1):
             continue
 
         if guard_size_oblivious(stride != expected_stride):
```

Comment:
you can do guard_or_true here and rename the function to is_non_overlapping_and_dense_or_false.

Reply:
its ok i  will address those in a different PR